### PR TITLE
Add gradient boosting decision tree algorithms

### PR DESCRIPTION
**Pull request by harupy**
Add the following gradient boosting decision tree algorithms:

- XGBoost
- LightGBM

### CODE REVIEW

**Comment by tkrabel** [comment]
@harupy thanks for the contribution :)

**Comment by harupy** [comment]
@tkrabel Thanks for merging!
